### PR TITLE
feat: add LPIPS and DISTS perceptual evaluation metrics

### PR DESCRIPTION
**Pull request by YeapJieShen**
⛔ **Do not merge yet — 2nd of a 6-PR stack.** Based on `feat/cli-subclass-mode`; merge that one first.

Adds LPIPS and DISTS as evaluation metrics, reported for the validation set and for each benchmark set.

## Why

An adversarial objective makes PSNR and SSIM **worse by design**, so the SRGAN work later in this stack would otherwise ship a training mode with no metric that tracks what it is optimising. LPIPS and DISTS correlate with human judgement better than PSNR does. Neither is a substitute for the paper's MOS study, and nothing here claims otherwise.

## What changed

- **`sisr/perceptual.py`** — `PERCEPTUAL_METRICS` (name → is-lower-better) and `perceptual_score(...)`. The one place either library's argument conventions are known.
- **`SREvalConfig`** — `perceptual_metrics: list[str] = []` and `lpips_net: Literal["alex","vgg","squeeze"] = "alex"`, plus a `perceptual_keys` property mirroring `psnr_keys`/`ssim_keys`.
- **`SRLightning._mean_perceptual`** — the single decision point for which perceptual metric this project computes, beside `_mean_ssim` for the same reason that one exists.
- **Tags** — `lpips/val` · `dists/val` · `lpips/{set}` · `dists/{set}`. No colorspace segment: both metrics are RGB-only by definition.
- **`_validate_monitor_metric`** — now also validates metric *direction*.
- **Packaging** — new `perceptual` extra (`lpips`); `torchmetrics` floor raised to `>=1.9`.

## Defaults are off, deliberately

`perceptual_metrics` is empty by default, so **every existing SRCNN/SRResNet run logs exactly the tags it logged before**. A test asserts no perceptual tag appears when unrequested.

## Two argument conventions, both silent-failure shaped

These are why the module exists as a seam rather than two inline calls:

1. **LPIPS needs `normalize=True`.** Its default `False` means "inputs are already `[-1,1]`"; this project's tensors are RGB `[0,1]`, so the default silently scores a squashed range — a plausible number, no error.
2. **The two disagree on reduction defaults** (LPIPS `'mean'`, DISTS `None`), so one would return a scalar and the other a per-image tensor under the same logging call. Both are passed explicitly.

Both are pinned by tests whose spies declare the parameters **without defaults**, so omitting either fails with `TypeError` rather than silently passing a wrong value.

## Monitor direction is now validated

Both checkpoint classes default to `mode="max"` — correct for PSNR/SSIM, exactly inverted for LPIPS/DISTS. `monitor_metric: lpips/val` at the default would have kept the **worst** checkpoint of the run, with nothing in the logs, tags or filenames revealing it. The validator now rejects the mismatch and names the mode to set instead. Direction is looked up per metric from `PERCEPTUAL_METRICS`, not restated.

All eight checkpoint entries across the three shipped templates monitor `psnr/val/RGB` or `ssim/val/RGB` at the default mode and continue to validate unchanged.

## Dependencies

**DISTS costs nothing** — torchmetrics implements it over torchvision's VGG16, both already required. **LPIPS needs the `lpips` package**, so it sits behind `pip install '.[perceptual]'` with an error naming the extra. The `torchmetrics>=1.9` floor is the version DISTS was *verified* present in here; it was not lowered to a version nobody tested.

## Testing

813 passed / 0 skipped / 0 failed. `ruff check` and `ruff format --check` clean.

**CI stays hermetic:** the torch hub cache is empty and CI has no network, so every test monkeypatches at a seam that is actually reached — none can trigger a weights download (~230 MB AlexNet / ~528 MB VGG16). Each new test ships with the mutation that proves it fails; one test that mocked a constant, and so was blind to the tensor choice it was named for, was caught in review and rebuilt to depend on its inputs.

## Stack

```
main
 └─ CLI subclass mode
     └─ THIS PR  · perceptual metrics (LPIPS/DISTS)
         └─ rolling checkpoints
             └─ SRGAN discriminator + adversarial loss
                 └─ SRGAN Lightning module + template
                     └─ docs
```

**Do not use "delete branch on merge"** — it races and closes the child PR instead of retargeting it.

### CODE REVIEW

**Review by YeapJieShen — COMMENTED** [review]
Human reviewed, all good.
